src/components/PopupWithForm.js

```diff
@@ -22,9 +23,31 @@ export default class PopupWithForm extends Popup {
     this._modalForm.addEventListener("submit", (evt) => {
       evt.preventDefault();
       const inputValues = this._getInputValues();
-      console.log("Input values before submit:", inputValues);
+      this.renderSaving(true);
       this._handleFormSubmit(inputValues);
+      this.renderSaving(false);
     });
     super.setEventListeners();
   }
+
+  stopSaving(popup) {
+    popup._modalForm.querySelector(".modal__save-button").textContent =
```

Comment:
` popup._modalForm`


You don't need to pass `popup` into the call because you have `this._modalForm` available here


```js


  stopSaving() {
    this._modalForm.querySelector(".modal__save-button").textContent =
      this._buttonText;
    this.close();
  }
```

Reply:
This is not working for me. I get an error message that this is undefined. Will look at this when I look at resolving the larger issue with 'this.'

Reply:
Fixed this.